docs/multi-target-framework.md

```diff
@@ -0,0 +1,150 @@
+# Multi-targeting
+
+## Developer needs / scenarios
+
+* Compile against an earlier version of our bindings.
+    * Example: a customer could be producing a NuGet targeting .NET 7, and
+      want to support the initial release of .NET 7. Said customer must then
+      be able to build against the bindings we shipped at the time.
+    * Ref: https://github.com/dotnet/sdk/issues/30103
+* Consume preview packages for a preview version of Xcode.
+    * Ref: https://github.com/xamarin/xamarin-macios/issues/18343
+
+## Developer experience
+
+### TargetFramework=net7.0-ios16.4
+
+Builds with the bindings we released for iOS 16.4 (Xcode 14.3).
+
+### TargetFramework=net7.0-ios
+
+Builds with the default bindings. This can change in any release (this is
+contrary to what other platforms do - specifically because older OS bindings
+might not work with newer Xcodes, and Apple auto-updates people's Xcodes, so
+it's rather frequent that customers use newer OS bindings. Having a lot of
+people specify the OS version in the target framework is undesirable).
+
+### TargetFramework=net8.0-ios17.0
+
+Builds with the bindings we've released for iOS 17.0 (on .NET 8).
+
+This might be preview bindings, in which case customers must also set the
+following property in their project file to make their intention clear:
+
+```xml
+<PropertyGroup>
+    <EnablePreviewFeatures>true</EnablePreviewFeatures>
```

Comment:
Will we also set this when building `Microsoft.iOS.dll`? Roslyn emits a custom attribute that tells you to add the property in your app roject.

Reply:
We're adding the `[RequiresPreviewFeatures]` when building `Microsoft.iOS.dll` using a beta version of Xcode:

https://github.com/xamarin/xamarin-macios/blob/8691e43c0a2de2b196fd64007ad5652abe0ba8c4/src/AssemblyInfo.cs.in#L42-L44